packages/editor/src/store/effects.js

```diff
@@ -54,6 +55,36 @@ import {
 	AUTOSAVE_POST_NOTICE_ID,
 } from './effects/posts';
 
+/**
+ * Block validity is a function of blocks state (at the point of a
+ * reset) and the template setting. As a compromise to its placement
+ * across distinct parts of state, it is implemented here as a side-
+ * effect of the block reset action.
+ *
+ * @param {Object} action RESET_BLOCKS action.
+ * @param {Object} store  Store instance.
+ *
+ * @return {?Object} New validity set action if validity has changed.
+ */
+export function validateBlocksToTemplate( action, store ) {
```

Comment:
So part of my purpose here in extracting this to a named function is to try to set a precedent for well-named / intentioned effect handlers. As you propose, we'd need a function which itself calls to `validateBlocksToTemplate`. It would likely start small, just extracting the blocks property from the action, but there'd be nothing to stop someone from adding to it, regressing to the poorer code standard.

Illustrated by code:

```js
// Bad:

RANDOM_ACTION( action, store ) {
	// Do A

	// Do B

	// Do C
}

Good:

RANDOM_ACTION: [
	doA( action, store ) {},
	doB( action, store ) {},
	doC( action, store ) {},
]
```

In general, I would have no issue with what you propose, and it'd probably make things even more clearly intentioned and easier to test. With needing the template and template lock details, it becomes a little more effort to be able to organize. 

I might poke around at it to see if I can find some happy compromise, but otherwise I'm not strongly inclined to change it.